pkg/controller/util.go

```diff
@@ -135,6 +135,13 @@ func newScratchPersistentVolumeClaimSpec(pvc *v1.PersistentVolumeClaim, pod *v1.
 			}
 		}
 	}
+	// When the original PVC is being handled by a populator, copy AnnSelectedNode to avoid issues with k8s scheduler
+	_, isPopulator := pvc.Annotations[cc.AnnPopulatorKind]
+	selectedNode := pvc.Annotations[cc.AnnSelectedNode]
+	if isPopulator && selectedNode != "" {
+		annotations[cc.AnnSelectedNode] = selectedNode
+	}
```

Comment:
maybe also put in a function? big chunk of code for this function which is very specific to populators :)
not a must though..

Reply:
hmmm I know I'm repeating some code here but I've tried some alternatives to reuse `SetNodeNameIfPopulator` and I think it looks uglier.